pkg/cloudprovider/providers/gce/cloud/context.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"context"
+	"time"
+)
+
+const (
+	defaultOperationTimeout = 1 * time.Hour
+)
+
+// DefaultContext returns a context with a default timeout, used for generated client operations.
+func DefaultContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), defaultOperationTimeout)
```

Comment:
Not sure how I feel about having such a small utility func. At the very least, I'd go with a name like `ContextWithOpTimeout`.
Will ask @bowei what he thinks.

Reply:
It looks ok to me for now. I would like to have this be tuneable of gce.conf and a method of GCE itself.

Let's get this in and do that change in a separate PR